### PR TITLE
chore(go-client): add generation thrift files of go-client

### PR DESCRIPTION
**Pull request by lengyuexuexuan**
### What problem does this PR solve? 
#1881

### What is changed and how does it work?

By uploading generation thrift files, the go client can be used directly by users through "go get" without the need to compile it locally.

##### Tests 

- Manual test (add detailed scripts or steps below)

### CODE REVIEW

**Comment by acelyc111** [comment]
@lengyuexuexuan Please check the failed CIs, for `Check License`, you need to add the new files to `.licenserc.yaml`.

**Comment by acelyc111** [comment]
Now that the go files have been pre-generated, the generate steps in https://github.com/apache/incubator-pegasus/blob/master/go-client/Makefile#L19 can be removed, right? Then you need to add the manual generate steps in `go-client/README.md`.

And more, you need to add steps in .github/workflows/lint_and_test_go-client.yml to check that the pre-generated go files are the same to the ones generated manually.

**Comment by lengyuexuexuan** [comment]
> Now that the go files have been pre-generated, the generate steps in https://github.com/apache/incubator-pegasus/blob/master/go-client/Makefile#L19 can be removed, right? Then you need to add the manual generate steps in `go-client/README.md`.
> 
> And more, you need to add steps in .github/workflows/lint_and_test_go-client.yml to check that the pre-generated go files are the same to the ones generated manually.


Is it possible to remove the thrift file directly in the .gitignore file, so that if the thrift file is modified, it can be uploaded directly?

**Comment by acelyc111** [comment]
> > Now that the go files have been pre-generated, the generate steps in https://github.com/apache/incubator-pegasus/blob/master/go-client/Makefile#L19 can be removed, right? Then you need to add the manual generate steps in `go-client/README.md`.
> > And more, you need to add steps in .github/workflows/lint_and_test_go-client.yml to check that the pre-generated go files are the same to the ones generated manually.
> 
> Is it possible to remove the thrift file directly in the .gitignore file, so that if the thrift file is modified, it can be uploaded directly?

Of course, feel free to do that.

**Review by acelyc111 — DISMISSED** [review]

**Comment by lengyuexuexuan** [comment]
> @lengyuexuexuan Do you have some time to improve the PR ? I'm pleased to give you some help if needed.

thanks. Today or Tomorrow I will improve this pr.

**Comment by lengyuexuexuan** [comment]
> @lengyuexuexuan Do you have some time to improve the PR ? I'm pleased to give you some help if needed.

thanks. Today or Tomorrow I will improve this pr.

**Comment by lengyuexuexuan** [comment]
@acelyc111 
Now I have uploaded the latest version of the thrift file, and remove go-client/idl/* in .gitignored file. 
I tested it and now users can directly pull the latest version of go client and use it directly. 
It needs another committer to aprrove this pr and  #1916.

**Comment by acelyc111** [comment]
according to the error, you need to add these file names [1] to `.licenserc.yaml` (add them in property place to keep in dictionary order).

1. https://github.com/apache/incubator-pegasus/actions/runs/9285106967/job/25548898496?pr=1917#step:3:141

**Comment by acelyc111** [comment]
@lengyuexuexuan please check this issue as well

https://github.com/apache/incubator-pegasus/actions/runs/9285106961/job/25848921042?pr=1917#step:6:4621

**Comment by lengyuexuexuan** [comment]
> @lengyuexuexuan please check this issue as well
> 
> https://github.com/apache/incubator-pegasus/actions/runs/9285106961/job/25848921042?pr=1917#step:6:4621

done.

File: .gitignore [hunk 1]
```diff
@@ -342,11 +342,11 @@ package-lock.json
...
-go-client/idl/admin/
-go-client/idl/cmd/
-go-client/idl/radmin/
-go-client/idl/replication/
-go-client/idl/rrdb/
+#go-client/idl/admin/
+#go-client/idl/cmd/
+#go-client/idl/radmin/
+#go-client/idl/replication/
+#go-client/idl/rrdb/
```
Comment (acelyc111) [inline .gitignore:349]:
```suggestion
```

File: .licenserc.yaml [hunk 2]
```diff
@@ -43,9 +43,12 @@ header:
     # Special files for golang.
     - '**/go.sum'
     # TODO(wangdan): Generated files for go client, could generate dynamically?
```
Comment (acelyc111) [inline .licenserc.yaml:45]:
```suggestion
```

**Review by acelyc111 — APPROVED** [review]

**Review by empiredan — APPROVED** [review]

**Pull request merged**